internal/cypress/v1/config.go

```diff
@@ -538,3 +540,32 @@ func (p *Project) IsSharded() bool {
 func (p *Project) GetAPIVersion() string {
 	return p.APIVersion
 }
+
+// GetSmartRetry returns the smartRetry config for the given suite.
+// Returns an empty config if the suite could not be found.
+func (p *Project) GetSmartRetry(suiteName string) config.SmartRetry {
+	for _, s := range p.Suites {
+		if s.Name == suiteName {
+			return s.SmartRetry
+		}
+	}
+	return config.SmartRetry{}
+}
+
+// FilterFailedTests filteres failed tests and sets to specified suite
```

Comment:
Need a bit more meat. It's already clear from the name that it's filtering tests. Any more insight to the developer as to the behavior? Any gotchas?

```suggestion
// FilterFailedTests takes the failed tests in the report and sets them as a test filter in the suite.
// The test filter remains unchanged if the report does not contain any failed tests.
```